City-of-Helsinki/berth-reservations-admin
# VEN-515 | Authentication flow

## Description :sparkles:
- Remove redux and its dependencies
- Implement Auth service based on `oidc-client`
- Persist the token in the local storage
- Remove the local schema
- Remove redundant components and files

## Issues :bug:

### Closes :no_good_woman:

### Related :handshake:

## Testing :alembic:

### Automated tests :gear:️
- https://github.com/City-of-Helsinki/berth-reservations-admin/blob/cd390ec4f6f9406ea1f53dd1ecf13221ad15a4c4/src/domain/auth/authService.ts#L1

### Manual testing :construction_worker_man:
- Click on the login button which redirects you to tunnistamo 
- After logging in tunnistamo should redirects you back to the page you were trying to access in the first place or the main page i.e `/harbors`
- Refresh the page manually and you should remain logged in
- On the header of the page, your name, fetched from the service you logged in through, appears on the right side
- Click on your name which displays a dropdown with a logout button
- Click on the logout button and you should be logged out

Advanced steps:
- Clear the local storage manually
- Try to navigate to any page the service should log you out and redirects you to the login page

## Screenshots :camera_flash:

## Additional notes :spiral_notepad:
